test/e2e/autoscale_test.go

```diff
@@ -268,3 +268,44 @@ func TestAutoscaleUpDownUp(t *testing.T) {
 	assertScaleDown(ctx)
 	assertScaleUp(ctx)
 }
+
+func assertScaleUpCountPods(ctx *testContext) {
+	ctx.logger.Infof("The autoscaler spins up additional replicas when traffic increases.")
+	// note: without the warm-up / gradual increase of load the test is retrieving a 503 (overload) from the envoy
+	// options: https://istio.io/docs/concepts/traffic-management/#timeouts-and-retries
+	err := generateTraffic(ctx, 20, 30*time.Second)
+	if err != nil {
+		ctx.t.Fatalf("Error during initial scale up: %v", err)
+	}
+	AssertNumberOfPods(ctx, 2)
```

Comment:
This is a good start (asserting the actual number of pods). I feel like we should probe much more often though (like once every second or two seconds?) to be able to rule out super flaky behavior by the autoscaler. I'm also not sure if we should assert for the exact number but a (small) range of pods (in this case 2/3 could be acceptable).

Probing more often could be done via a similar mechanism than `DiagnoseMe`, which would check for a given range of pods in between `assertNumberOfPods` calls. For this example, it'd be 1-3, then 2-4 and so on.

Does that make sense?

Reply:
@markusthoemmes thank you for the feedback. yes, I think the probing and beeing a bit more relax in assertion makes sense.  I've adopted the comments and run the test multiple times locally. Are the changes reflecting your suggestions?

One other aspect I was not sure is whether we should test against the "Replicas" (wanted) vs the "ReadyReplicas" (have) of the deployment. Currently the test is checking against "wanted", should we check against both?